xdsl/dialects/arith.py

```diff
@@ -352,6 +371,24 @@ def py_operation(lhs: int, rhs: int) -> int | None:
     def is_right_unit(attr: IntegerAttr) -> bool:
         return attr.value.data == 0
 
+    def fold(self):
+        lhs = self.get_constant(self.lhs)
+        rhs = self.get_constant(self.rhs)
+        if lhs is not None and rhs is not None:
+            assert isinstance(lhs, IntegerAttr)
+            assert isinstance(rhs, IntegerAttr)
+            assert (type := lhs.get_type()) == rhs.get_type()
+            assert (
+                result := self.py_operation(lhs.value.data, rhs.value.data)
+            ) is not None
+            # TODO: Argument of type "Attribute" cannot be assigned to parameter "value_type"
+            # of type "IntCovT@__init__" in function "__init__"
+            return (IntegerAttr(result, type),)
+        if isinstance(lhs, IntegerAttr) and lhs.value.data == 0:
+            return (self.rhs,)
+        if isinstance(rhs, IntegerAttr) and rhs.value.data == 0:
+            return (self.lhs,)
```

Comment:
In MLIR, only `x+0 --> x` is checked, not the other direction. I'm not sure why that's the case.

Reply:
https://mlir.llvm.org/docs/Canonicalization/

Reply:
I guess you're right 👍